Feature/log2video

m3d opened the pull request:
Tool for video conversion from log file. Cherry-picked commits from "feature/eduro" branch.

I would squash the commits and maybe already move them, as discussed over release mail, to "osgar/tools", OK?

zwn commented:
Should this be inside `osgar` package? If it should, let's handle gracefully missing opencv by wrapping the `import cv2` with try/except with some nice error message explaining what to do to make this work (opencv is not part of dependencies of the package osgar).

Or are you thinking about a separate osgar-tools package?

m3d commented:
I like the idea of extra `osgar-tools` package supporting cv2, matplot, pygame and maybe other "heavy" libraries used for analysis. So do you propose to keep `tools` directory as it is now?

zwn commented:
> So do you propose to keep `tools` directory as it is now?

No. I suppose that would mean making a pypi released `osgar-tools` containing `tools` python package which would be a bad idea. Having such a generic sounding package name at the top of the namespace does not sound appropriate.

I propose moving it inside `osgar` directory and packaging it as `osgar-tools` package on pypi. That way, even after installing, it would be obvious it is part of osgar.

m3d commented:
OK, log2video.py is now moved to osgar/tools + added mini README.md for newly created direcotry/package. I added also try/except for import cv2, but I am not sure what about numpy?? BTW it can be necessary to have OpenCV+numpy even in regular osgar for image processing.

zwn commented:
I think all imports of packages that are not accounted for in the dependencies in setup.py should be wrapped with try/except. For the users of the packages from pypi, I think it is appropriate.

So, if log2video.py is part of pypi package osgar-tools which depends on opencv and numpy, then we do not need the try/except (but it does not hurt).

Does it make sense?

m3d commented:
OK, fixed - I will rebase, merge and there will be another review with extra print time ... something like in:
https://youtu.be/6yNynVL6FR4
